### ISSUE TITLE
Custom Stoploss explanation

### ISSUE THREAD

**Issue by quantsirius** [freqtrade/freqtrade#6955]
* Operating system: Ubuntu
  * Python Version: Python 3.8.10
  * CCXT version: ccxt==1.83.62
  * Freqtrade Version: freqtrade 2022.5
  
There are some questions about custom stoploss works that doesn't clear 
It is a result of the simple strategy without custom stoploss

![image](https://user-images.githubusercontent.com/67390955/172834667-fe09a5d1-c6c1-4147-b74a-1cbe1470bf4e.png)
 
Then only added "use_custom_stoploss=True" without any changes. No any custom stoploss rules has not been added. 

    # This attribute will be overridden if the config file contains "minimal_roi".
    minimal_roi = {
        # "60": 0.01,
        # "30": 0.02,
        "0": 0.03
    }

    # Optimal stoploss designed for the strategy.
    # This attribute will be overridden if the config file contains "stoploss".
    stoploss = -0.03

    use_custom_stoploss=True

    # Trailing stoploss
    trailing_stop = False
    # trailing_only_offset_is_reached = False
    # trailing_stop_positive = 0.01
    # trailing_stop_positive_offset = 0.0  # Disabled / not configured

    # Run "populate_indicators()" only for new candle.
    process_only_new_candles = False

![image](https://user-images.githubusercontent.com/67390955/172835560-c68d82ee-c59c-4214-a262-785b9817545a.png)

if you don't mind explaining, I would like to know how it exactly work.
 
Another question. In prior i read Stoploss is adjusted for leverage. Is it custom stoploss adjusted also?

**Comment by xmatthias** [comment]
the default for the `custom_stoploss()` callback  is to return the strategy's stoploss.

as the callback is called [once every iteration](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) - it will act as a trailing stoploss (trailing with the initial stoploss value - 3% in your case).

It's "almost" identical to running a specific trailing stop combination - but with some limitation (and hence differences) as backtesting can't take certain assumptions on how the trailing stop triggered.

custom stoploss is just a different way to define the stoploss - it's behavior itself will not change, so leverage adjustment applies to that too.

**Comment by quantsirius** [comment]
> the default for the `custom_stoploss()` callback is to return the strategy's stoploss.
> 
> as the callback is called [once every iteration](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) - it will act as a trailing stoploss (trailing with the initial stoploss value - 3% in your case).
> 
> It's "almost" identical to running a specific trailing stop combination - but with some limitation (and hence differences) as backtesting can't take certain assumptions on how the trailing stop triggered.

OK

> custom stoploss is just a different way to define the stoploss - it's behavior itself will not change, so leverage adjustment applies to that too.

I understand the risk with leverage but it is mean trailing stoploss from indicator (like ATR from docs) is useless by default.
And if want to use it for 2x leverage need to multiply stoploss*2 like

```
# Use parabolic sar as absolute stoploss price
stoploss_price = last_candle['sar']

# Convert absolute price to percentage relative to current_rate
if stoploss_price < current_rate:
        return ((stoploss_price / current_rate) - 1) * 2
```

or is there exist better way?

**Comment by xmatthias** [comment]
as the [documentation states](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) - the return value is not a price, but the relative offset from the current price - calculated against your investment (so stoploss = capital at risk).

What you're talking about is the % of pair/price movements - however stoploss is a features to limit / contain risk.
As such - a stoploss of 10% should never cause more than 10% loss.

Would stoploss work differently (based on the pair movement, not the capital), then a stoploss of 10% with 2x leverage would result in a 20% loss - or 50% loss if 5x leverage would be used (or 100% loss (in that case a liquidation) in a 10x leverage case).

Assume you invested 100\$ (the position is 200\$) - you'd be down to 80$ with this 1 10% 2x stoploss - even though you clearly stated that your maximum acceptable risk for that trade is 10% from your invested capital (in that trade).

**Comment by quantsirius** [comment]
> as the [documentation states](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) - the return value is not a price, but the relative offset from the current price - calculated against your investment (so stoploss = capital at risk).
> 
> What you're talking about is the % of pair/price movements - however stoploss is a features to limit / contain risk. As such - a stoploss of 10% should never cause more than 10% loss.
> 
> Would stoploss work differently (based on the pair movement, not the capital), then a stoploss of 10% with 2x leverage would result in a 20% loss - or 50% loss if 5x leverage would be used (or 100% loss (in that case a liquidation) in a 10x leverage case).
> 
> Assume you invested 100) - you'd be down to 80$ with this 1 10% 2x stoploss - even though you clearly stated that your maximum acceptable risk for that trade is 10% from your invested capital (in that trade).

I completely understand that. Just want use custom stoploss by ATR as trailing exit. Default stoploss would be still limit maximum risk. Сan I do it like above?

**Comment by xmatthias** [comment]
i guess multiplying with leverage will work - but i'd use `trade.leverage` instead of 2 (that way it'll scale should you decide to change the leverage at a later point (or dynamically).

**Comment by jxbzwnqv** [comment]
thank you

**Comment by amjad-12** [comment]
I want to ask also please 
I am using adjust trade position and in the first position my stake_amount is 10 and in the first adjust position I retrun stake_amount *2 and in the second adjusting I return stake_amount * 3 and also I am using the custom_stoploss and I check in it if the trade.nr_of_successfultrade equal to 3 I return 0.01 so I expect this value to be for the second price adjusting and only my lose be -1% but what I had is losing 6% so hwo to use it to make the 1% for the last price buying

**Comment by amjad-12** [comment]
and what I want is like to use the stoploss_from_open but for the second adjusting price not the open price
and I see this in the docs 
        return stoploss_from_absolute(current_rate + (side * candle['atr'] * 2), 
                                      current_rate, is_short=trade.is_short,
                                      leverage=trade.leverage)

but I don't know exactly how to use it and I dont use the laverage but I stored the third price I buy on it when buying so shuld I replace the current_rate + (side * candle['atr'] * 2) with it and it will be done? 

also my trade is spot so how i should set the parameters in it

**Comment by stash86** [comment]
Please open new issue instead of commenting on old issue
